skimage/io/sift.py

```diff
@@ -62,9 +60,10 @@ def _sift_read(f, mode='SIFT'):
 
     data = np.fromfile(f, sep=' ')
     if data.size != nr_features * datatype.itemsize / np.dtype(float).itemsize:
-        raise IOError("Invalid %s feature file." % mode)
+        raise IOError("Invalid {} feature file.".format(mode))
 
-    f.close()
+    if f_is_str:
+        f.close()
```

Comment:
Ah. Well then I have three non-mutually suggestions:

- don't overwrite `f`. Change the argument name to `filelike`, and then rename the open file to `file`.
- rename `f_is_str` to `filelike_was_str`
- add a comment next to this if clause with some version of this explanation